onepassword/items.go

```diff
@@ -2,10 +2,36 @@ package onepassword
 
 import (
 	"encoding/json"
+	"fmt"
 	"strings"
 	"time"
 )
 
+type ItemState string
+
+const (
+	Active   ItemState = ""
+	Deleted  ItemState = "DELETED"
+	Archived ItemState = "ARCHIVED"
+	Unknown  ItemState = "UNKNOWN"
+)
+
+// UnmarshalJSON Unmarshall Item State enum strings to Go string enums
+func (is *ItemState) UnmarshalJSON(b []byte) error {
+	var s string
+	json.Unmarshal(b, &s)
+	state := ItemState(s)
+	switch state {
+	case Deleted, Archived:
+		*is = state
+		return nil
+	default:
+		*is = Unknown
+	}
+
+	return fmt.Errorf("%q isn't a recognized item state", s)
+}
+
```

Comment:
Do we need custom unmarshalling logic? Isn't that covered by `ItemState` being a `string`?

I see it also does validation, but we're currently not doing that for any other API type.

Reply:
Yeah, it covers it. I agree. I've removed it